services/app-web/src/components/Form/FieldPreserveScrollPosition/FieldPreserveScrollPosition.tsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react'
+import { useFormikContext } from 'formik'
+
+/**
+ * This is a renderless component
+ *
+ * It relies on the Formik context to work, so it must ALWAYS be rendered
+ * inside of a Formik form.
+ *
+ * For typescript -  make sure to pass in typed props to allow component to properly infer formik context types.
+ */
+
+export function FieldPreserveScrollPosition<T>({
+    fieldName,
+}: {
+    fieldName: keyof T
+}): null {
+    const { values } = useFormikContext<T>()
+    const fieldValue = values[fieldName]
+    useEffect(() => {
+        document.querySelector(`#${fieldName}`)?.scrollIntoView()
+    }, [fieldValue, fieldName])
```

Comment:
Should we use useLayoutEffect here also?

Reply:
Yes, good call. I added it in [185b630](https://github.com/CMSgov/managed-care-review/pull/935/commits/185b63036ca33f267763e1e1ee1419c07d9c5b71) let me know if you see any different in this commit.

Reply:
Looks great! There was an issue the first round of testing, where going back to the `Contract Details` page would scroll to the fields, this is no longer happening. I wonder if changing this to `useLayoutEffect` from `useEffect` fix that because, I think `useEffect` fires after `useLayoutEffect`.

Reply:
I bet you are right!